### PR TITLE
Azure Blob Storage: Support endpoint & secondary endpoint params

### PR DESCRIPTION
**Pull request by amotl**
## About

This patch adds support for private endpoints to the Azure Storage subsystem.

## References
```
@pytest.mark.skip(reason="CrateDB does not support custom endpoints for Azure Blob Storage")
```
-- https://github.com/crate-workbench/cratedb-retention/blob/0dfe1cb1f5/tests/test_cli.py#L314-L319

## Checklist

 - [x] Added an entry in `CHANGES.txt` for user facing changes
 - [x] Updated documentation & `sql_features` table for user facing changes
 - [x] Touched code is covered by tests
 - [x] [CLA](https://crate.io/community/contribute/cla/) is signed
 - [x] This does not contain breaking changes, or if it does:
    - It is released within a major release
    - It is recorded in ``CHANGES.txt``
    - It was marked as deprecated in an earlier release if possible
    - You've thought about the consequences and other components are adapted
      (E.g. AdminUI)

### CODE REVIEW

**Comment by matriv** [comment]
Keep in mind that `endpoint` and `secondary_endpoint` are not masked for `sys.repositories` as `endpoint_suffix` is, and I don't see myself a reason why they should be masked.

Additionally, I'm not sure why `endpoint_suffix` is masked. Any thoughts?

**Comment by matriv** [comment]
retest this please

File: docs/sql/statements/create-repository.rst [hunk 1]
```diff
@@ -418,13 +418,37 @@ Parameters
...
+
+**endpoint**
+  | *Type:*    ``text``
+
+  The Azure Storage account endpoint suffix.
+
+  .. NOTE::
+
+      ``endpoint`` cannot be used in combination with
+      `sql-create-repo-azure-endpoint_suffix`_.
```
Comment (amotl) [inline docs/sql/statements/create-repository.rst:438]: Thank you for outlining those details.

File: docs/sql/statements/create-repository.rst [hunk 2]
```diff
@@ -418,13 +418,37 @@ Parameters
...
+
+**secondary_endpoint**
+  | *Type:*    ``text``
+
+  The Azure Storage account secondary endpoint.
+
+  .. NOTE::
+
+      ``secondary_endpoint`` cannot be used in if
+      `sql-create-repo-azure-endpoint`_ is not specified.
```
Comment (amotl) [inline docs/sql/statements/create-repository.rst:450]:
```suggestion
      ``secondary_endpoint`` cannot be used if
      `sql-create-repo-azure-endpoint`_ is not specified.
```

File: plugins/es-repository-azure/src/test/java/org/elasticsearch/repositories/azure/AzureSnapshotIntegrationTest.java [hunk 3]
```diff
@@ -41,7 +41,8 @@
 
 import io.crate.testing.Asserts;
 
-public class AzureSnapshotIntegrationTest extends IntegTestCase {
+@IntegTestCase.ClusterScope(scope = IntegTestCase.Scope.TEST)
+    public class AzureSnapshotIntegrationTest extends IntegTestCase {
```
Comment (mfussenegger) [inline plugins/es-repository-azure/src/test/java/org/elasticsearch/repositories/azure/AzureSnapshotIntegrationTest.java:45]:
```suggestion
public class AzureSnapshotIntegrationTest extends IntegTestCase {
```

File: plugins/es-repository-azure/src/test/java/org/elasticsearch/repositories/azure/AzureSnapshotIntegrationTest.java [hunk 4]
```diff
@@ -104,6 +105,48 @@ public void create_azure_snapshot_and_restore_it() {
...
 
+    @Test
+    public void create_azure_snapshot_and_restore_with_secondary_endpoint() {
+        execute("CREATE TABLE t1 (x int)");
+        assertThat(response.rowCount()).isEqualTo(1L);
+
+        var numberOfDocs = randomLongBetween(0, 10);
+        for (int i = 0; i < numberOfDocs; i++) {
+            execute("INSERT INTO t1 (x) VALUES (?)", new Object[]{randomInt()});
+        }
```
Comment (mfussenegger) [inline plugins/es-repository-azure/src/test/java/org/elasticsearch/repositories/azure/AzureSnapshotIntegrationTest.java:116]:
```suggestion
        int numberOfDocs = randomIntBetween(0, 10);
        Object[][] rows = new Object[numberOfDocs][];
        for (int i = 0; i < numberOfDocs; i++) {
            rows[i] = new Object[] { randomInt() };
        }
        execute("INSERT INTO t1 (x) VALUES (?)", rows);
```

Bulk inserts are faster

File: plugins/es-repository-azure/src/test/java/org/elasticsearch/repositories/azure/AzureSnapshotIntegrationTest.java [hunk 5]
```diff
@@ -121,4 +164,9 @@ private String httpServerUrl() {
         InetSocketAddress address = httpServer.getAddress();
         return "http://" + InetAddresses.toUriString(address.getAddress()) + ":" + address.getPort();
     }
+
+    private String invalidHttpServerUrl() {
+        InetSocketAddress address = httpServer.getAddress();
+        return "http://invalidHost:" + address.getPort();
```
Comment (mfussenegger) [inline plugins/es-repository-azure/src/test/java/org/elasticsearch/repositories/azure/AzureSnapshotIntegrationTest.java:170]:
```suggestion
        return "http://dummy.invalid:" + address.getPort();
```

See https://www.rfc-editor.org/rfc/rfc2606#page-2

**Review by mfussenegger — APPROVED** [review]

File: plugins/es-repository-azure/src/test/java/org/elasticsearch/repositories/azure/AzureSnapshotIntegrationTest.java (hunk 4 above)
Comment (matriv) [reply plugins/es-repository-azure/src/test/java/org/elasticsearch/repositories/azure/AzureSnapshotIntegrationTest.java]: thx!

File: plugins/es-repository-azure/src/test/java/org/elasticsearch/repositories/azure/AzureSnapshotIntegrationTest.java (hunk 5 above)
Comment (matriv) [reply plugins/es-repository-azure/src/test/java/org/elasticsearch/repositories/azure/AzureSnapshotIntegrationTest.java]: thx!

**Comment by matriv** [comment]
retest this please

**Comment by matriv** [comment]
retest this please

**Pull request merged**